doc/metadata_routing.rst

```diff
@@ -116,26 +129,27 @@ that :func:`~model_selection.cross_validate` does not pass the weights along::
   ...     scoring=weighted_acc,
   ... )
 
-If :meth:`linear_model.LogisticRegressionCV.set_fit_request` has not
-been called, :func:`~model_selection.cross_validate` will raise an
-error because ``sample_weight`` is passed in but
-:class:`~linear_model.LogisticRegressionCV` would not be explicitly configured
-to recognize the weights.
+If :meth:`linear_model.LogisticRegressionCV.set_fit_request` had not been called,
+:func:`~model_selection.cross_validate` would raise an error because ``sample_weight``
+is passed but :class:`~linear_model.LogisticRegressionCV` would not be explicitly
+configured to recognize the weights.
 
 Unweighted feature selection
 ----------------------------
 
-Setting request values for metadata are only required if the object, e.g. estimator,
-scorer, etc., is a consumer of that metadata Unlike
-:class:`~linear_model.LogisticRegressionCV`, :class:`~feature_selection.SelectKBest`
-doesn't consume weights and therefore no request value for ``sample_weight`` on its
-instance is set and ``sample_weight`` is not routed to it::
+Routing metadata is only possible if the object's method knows how to use the metadata,
+meaning they have to have it as a parameter. Only then we can set request values for
+metadata using `set_fit_request(sample_weight=True)`, for instance. This makes the
+object a :term:`consumer <consumer>`.
```

Comment:
strictly speaking this is not true. Estimators can configure having a parameter as a consumer w/o having it as an explicit arg in the signature of the method. But I don't mind having this as you changed if it's more clear.

Reply:
Yes, they could be accepting `**kwarg` and explicitly setting request values for some of those args for the machinery to work. They don't have to be explicit parameters.